app/controllers/teams_controller.rb

```diff
@@ -0,0 +1,100 @@
+class TeamsController < ApplicationController
+  skip_before_filter :authenticate_user!, only: [:switch, :new, :create]
+  skip_after_filter :verify_authorized, only: [:switch, :new, :create]
+  skip_after_filter :verify_policy_scoped, only: [:switch, :new]
+
+  def switch
+    if current_user
+      team = current_user.teams.friendly.find(params[:id])
+      session[:current_team_slug] = team.slug
+    else
+      team = Team.not_archived.friendly.find(params[:id])
+      session[:team_slug] = team.slug
+    end
+    redirect_to root_path
+  end
+
+  # GET /teams/new
+  # GET /teams/new.xml
+  def new
+    @team = Team.new
+
+    respond_to do |format|
+      format.html # new.html.erb
+      format.xml  { render xml: @team }
+    end
+  end
+
+  def show
+    redirect_to edit_team_path(current_team)
+  end
+
+  # GET /teams/1/edit
+  def edit
+    @team = current_team
+    authorize @team
+  end
+
+  # POST /teams
+  # POST /teams.xml
+  def create
+    @team = Team.new(allowed_params)
+
+    respond_to do |format|
+      if verify_recaptcha && ( @team = TeamOperations::Create.(@team, current_user) )
+        format.html do
+          session[:team_slug] = @team.slug
+          flash[:notice] = t('teams.team was successfully created')
+          redirect_to(root_path)
+        end
+        format.xml  { render xml: @team, status: :created, location: @team }
```

Comment:
huahua, dunno. Just kept what was already there :-D (copied from the update method, just to keep the same behavior)